datahub-web-react/src/app/permissions/roles/ManageRoles.tsx

```diff
@@ -0,0 +1,262 @@
+        setPage(newPage);
+    };
+
+    const tableColumns = [
+        {
+            title: 'Name',
+            dataIndex: 'name',
+            key: 'name',
+            render: (_, record: any) => {
+                return (
+                    <>
+                        <RoleName
+                            onClick={() => onViewRole(record.role)}
+                            style={{ color: record?.editable ? '#000000' : '#8C8C8C' }}
+                        >
+                            {record?.name}
+                        </RoleName>
+                    </>
+                );
+            },
+        },
+        {
+            title: 'Description',
+            dataIndex: 'description',
+            key: 'description',
+            render: (description: string) => description || '',
+        },
+        {
+            title: 'Users',
+            dataIndex: 'users',
+            key: 'users',
+            render: (_: any, record: any) => {
+                return (
+                    <>
+                        <AvatarsGroup
+                            users={record?.users}
+                            groups={record?.resolvedGroups}
+                            entityRegistry={entityRegistry}
+                            maxCount={3}
+                            size={28}
+                        />
+                    </>
+                );
+            },
+        },
+        {
+            dataIndex: 'add_users',
+            key: 'add_users',
+            render: (_: any, record: any) => {
+                return (
+                    <Tooltip title={`Assign ${record.name} role to users`}>
+                        <Button
+                            onClick={() => {
+                                setIsBatchAddRolesModalVisible(true);
+                                setFocusRole(record.role);
+                            }}
+                        >
+                            ADD USERS
+                        </Button>
+                    </Tooltip>
+                );
+            },
+        },
+    ];
+
+    const tableData = roles?.map((role) => ({
+        role,
+        urn: role?.urn,
+        type: role?.type,
+        description: role?.description,
+        name: role?.name,
+        users: role?.relationships?.relationships
+            .filter((relationship) => relationship?.entity?.__typename === 'CorpUser')
+            .map((relationship) => relationship.entity as CorpUser),
+        policies: role?.relationships?.relationships
+            .filter((relationship) => relationship?.entity?.__typename === 'DataHubPolicy')
+            .map((relationship) => relationship.entity as DataHubPolicy),
+    }));
+
+    return (
+        <PageContainer>
+            {rolesLoading && !rolesData && (
+                <Message type="loading" content="Loading roles..." style={{ marginTop: '10%' }} />
+            )}
+            {rolesError && message.error('Failed to load roles :(')}
```

Comment:
Please use the new format that I mentioned above... We are no longer using these weak :( error messages..

Reply:
Done!